audoma/drf/fields.py

```diff
@@ -144,3 +145,60 @@ def __init__(self, choices, **kwargs):
         else:
             self.original_choices = choices
         super().__init__(choices, **kwargs)
+
+
+class SerializerMethodField(ExampleMixin, fields.Field):
+    def _parse_field(self, field):
+        if field is None:
+            return None
+        elif isinstance(field, fields.Field):
+            return field
+        else:
+            raise ValueError(
+                f"Incorrect type of field, field \
+                    must be an instance of rest_framework.fields.Field.\
+                        Passed value: {field}"
+            )
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.method_name = kwargs.pop("method_name", None)
+        self.field = self._parse_field(kwargs.pop("field", None))
+        is_writable = kwargs.pop("is_writable", False)
+        if is_writable and self.field is None:
+            raise ValueError("Writable SerializerMethodField must have field defined.")
+
+        kwargs["source"] = "*"
+        kwargs["read_only"] = not is_writable
+        super().__init__(*args, **kwargs)
+
+    def __getattribute__(self, __name: str) -> Any:
+        value = super().__getattribute__(__name)
```

Comment:
I don't think it should be always run...

Reply:
Why it should not be run always?